ra/registration-authority.go

```diff
@@ -110,6 +111,12 @@ func validateEmail(ctx context.Context, address string, resolver bdns.DNSResolve
 	}
 	splitEmail := strings.SplitN(address, "@", -1)
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
+	if len(splitEmail) > 2 {
```

Comment:
@jsha @rolandshoemaker thanks for your patience! I updated the PR to use `mail.ParseAddressList` and check if the length is greater than 1, as well as updated `multipleAddressDetail` to match what we're checking for. Hope this is closer to what you're looking for!